flaskapp/models/__init__.py

```diff
@@ -0,0 +1,54 @@
+from flaskapp import db
+from flaskapp.models.cards import Card, CardType, CardSubType
+from sqlalchemy.orm.exc import NoResultFound
+import json
+
+# created this folder (aka module)
+# split it into users and cards
+# all user tables are in the right file
+# all card related things are in the card file
+
+def get_or_create(type_name, ModelClass):
+    try:
+        return ModelClass.query.filter_by(name=type_name).one()
+    except NoResultFound, e:
+        card_type = ModelClass(name=type_name)
+        db.session.add(card_type)
+        return card_type
+
+def upload_some_cards():
+    json_file = open('mtgcard.json')
+    carddata = json.load(json_file)
+    # firstItem = carddata[0]
+    # secondItem = carddata[1]
+    # newListTest = []
+    # newListTest.append(firstItem)
+    # newListTest.append(secondItem)
+    # print(firstItem['imageUrl'])
+    
+    cardList = []
+
+    # temp set to verify if the card shows up twice in the json
+    added = set()
+    for card in carddata:
+        if card['name'] in added:
+            continue
+        print card['name']
+        added.add(card['name'])
+        new_entry = Card(
+            name=card['name'],
+            power=card.get('power'),
+            toughness=card.get('toughness'),
+            manaCost=card['manaCost'],
+            image=card.get('imageUrl'),
+            card_types=map(lambda x: get_or_create(x, CardType), card['types']),
+            card_sub_types=map(lambda x: get_or_create(x, CardSubType), card['subtypes'])
```

Comment:
we will have to go over this part

Reply:
ah yeah, not exactly clear what's happening here, I'll make it more clean.

Reply:
I'll write this in a few different, equal ways

**version with map:**
```python
card_types = map(lambda x: get_or_create(x, CardType), card['types'])```

**naive for loop version**
```python
card_types = []
for type_name in card['types']:
     _type = get_or_create(type_name, CardType) # could be CardSubType, since they're very similar
     card_types.append(_type)
```
**Finally, the more elegant: list comprehension:**
```python
card_types = [get_or_create(type_name, CardType) for type_name in card['types']]
```